### ISSUE TITLE
Support multiple return values

### ISSUE THREAD

**Issue by maximiliankaul** [Fraunhofer-AISEC/cpg#1089]
The CPG should support multiple return values per statement. This is useful for language constructs like `a, b = (1, 2)` or `value, error := someFunction()`.
Note, that such constructs can also happen in unexpected places like loops `for a, b, c in ...` where we would have three loop variables in this example.

**Comment by KuechA** [comment]
The expressions supporting multiple values will model the targets with lists.

Tasks:
* [ ] Replace the `=` Binary Operator with a dedicated `AssignmentExpression` class.
* [ ] Adapt the `variable` in the `ForEachStatement`.
* [ ] The support for multiple return types/values of functions has to be expanded (see TODOs in code)

Tricky problems:
* (1) `a, b = foo()`: The function call should be modeled only once and magically be "expanded"
* (2) `int i, a, b = 1;`: Not all declared variables are initialized
* (3) Various ways to initialize a variable in C++ should be represented in the same way. The different options should probably be kept in some properties.
* (4) Mixture of `DeclaredReferenceExpressions` and `VariableDeclarations` (e.g. `a` is an already defined variable and used in `a, b = 1,2`): This probably needs handling refs and decls at the same time
* (5) It should be easy to identify uninitialized variables (= write a query)

Discussed solutions:
1) Split multiple assignments into multiple nodes. Problems: 1
2) Assignment statement with lists for `lhs`, `rhs`, `declarations`. Problems: 1, 2 (keeping the lists synchronized). Needs to change the VariableDeclaration to get rid of the initializer

**Comment by peckto** [comment]
If we treat multiple return values as a container, we could add an implicit "unpack" node in situations where the return values are unpacked to separate values.
The "unpack" functionality is probably language specific and could be implemented in separate pass.

**Comment by oxisto** [comment]
@maximiliankaul Is this issue still relevant? I think we implemented this in python using our assign expression?

**Comment by maximiliankaul** [comment]
> @maximiliankaul Is this issue still relevant? I think we implemented this in python using our assign expression? Or at least it should be done once #1729 is merged.

I think it's no longer relevant. I'll double check that we have tests for all the corner cases discussed above and then close this issue asap.

**Comment by maximiliankaul** [comment]
There are still some bugs. See #1807 for fixes.
